src/network.ts

```diff
@@ -0,0 +1,155 @@
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import { fs } from 'mz';
+import * as networkCheck from 'network-checker';
+import * as os from 'os';
+import * as url from 'url';
+
+import * as constants from './lib/constants';
+import { EEXIST } from './lib/errors';
+import { checkTruthy } from './lib/validation';
+
+import blink = require('./lib/blink');
+
+const networkPattern = {
+	blinks: 4,
+	pause: 1000,
+};
+
+let isConnectivityCheckPaused = false;
+let isConnectivityCheckEnabled = true;
+
+function checkHost(
+	opts: networkCheck.ConnectOptions,
+): boolean | PromiseLike<boolean> {
+	if (!isConnectivityCheckEnabled || isConnectivityCheckPaused) {
+		return true;
+	}
+	return networkCheck.checkHost(opts);
+}
+
+function customMonitor(
+	options: networkCheck.ConnectOptions,
+	fn: networkCheck.MonitorChangeFunction,
+) {
+	return networkCheck.monitor(checkHost, options, fn);
+}
+
+export function enableCheck(enable: boolean) {
+	isConnectivityCheckEnabled = enable;
+}
+
+async function vpnStatusInotifyCallback(): Promise<void> {
+	try {
+		await fs.lstat(constants.vpnStatusPath);
+		isConnectivityCheckPaused = true;
+	} catch {
+		isConnectivityCheckPaused = false;
+	}
+}
+
+export const startConnectivityCheck = _.once(
+	async (
+		apiEndpoint: string,
+		enable: boolean,
+		onChangeCallback?: networkCheck.MonitorChangeFunction,
+	) => {
+		enableConnectivityCheck(enable);
+		if (!Boolean(apiEndpoint)) {
+			console.log('No API endpoint specified, skipping connectivity check');
+			return;
+		}
+
+		Bluebird.resolve(fs.mkdir(constants.vpnStatusPath))
+			.catch(EEXIST, () => {
+				console.log('VPN status path exists.');
+			})
+			.then(() => {
+				fs.watch(constants.vpnStatusPath, vpnStatusInotifyCallback);
+			});
+
+		if (enable) {
+			vpnStatusInotifyCallback();
+		}
+
+		const parsedUrl = url.parse(apiEndpoint);
+		const port = parseInt(parsedUrl.port!, 10);
+
+		customMonitor(
+			{
+				host: parsedUrl.hostname,
+				port: port || (parsedUrl.protocol === 'https' ? 443 : 80),
+				path: parsedUrl.path || '',
+				interval: 10 * 1000,
+			},
+			connected => {
+				if (onChangeCallback != null) {
+					onChangeCallback(connected);
+				}
+				if (connected) {
+					console.log('Internet Connectivity: OK');
+					blink.pattern.stop();
+				} else {
+					console.log('Waiting for connectivity...');
+					blink.pattern.start(networkPattern);
+				}
+			},
+		);
+	},
+);
+
+export function enableConnectivityCheck(enable: boolean) {
+	enable = checkTruthy(enable) || true;
```

Comment:
This is equivalent to `enable = true` atm, `x ? y` and `x || y` are very different beasts and we generally only use `x ? y` when we specifically need that behavior